### PR TITLE
BIGINT overflow checks

### LINKED ISSUE #4523: Make queries that contain BIGINT overflows fail

### PR DESCRIPTION
**Pull request by maciejgrzybek**
This fixes: https://github.com/facebook/presto/issues/4523

### CODE REVIEW

File: presto-main/src/test/java/com/facebook/presto/sql/gen/TestExpressionCompiler.java [hunk 1]
```diff
@@ -282,7 +292,7 @@ public void testBinaryOperatorsBoolean()
     public void testBinaryOperatorsLongLong()
             throws Exception
     {
-        for (Long left : longLefts) {
+        for (Long left : smallLongs) {
```
Comment (cawallin) [inline presto-main/src/test/java/com/facebook/presto/sql/gen/TestExpressionCompiler.java:294]: why not test the extreme long values here?

**Comment by cawallin** [comment]
If possible, maybe the parsing fix should go in before this does, so that the tests don't need to be commented out? This commit itself looks good % my comment about not removing unit tests

File: presto-main/src/main/java/com/facebook/presto/operator/aggregation/LongSumAggregation.java [hunk 2]
```diff
@@ -31,7 +34,12 @@ private LongSumAggregation() {}
     public static void sum(NullableLongState state, @SqlType(StandardTypes.BIGINT) long value)
     {
         state.setNull(false);
-        state.setLong(state.getLong() + value);
+        try {
+            state.setLong(Math.addExact(state.getLong(), value));
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/operator/aggregation/LongSumAggregation.java]: Add a helper "addExact" that does the Math call and translates the exception rather than copying this everywhere

Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/operator/aggregation/LongSumAggregation.java]: Nevermind, I see they're all different. Though this call site could call the Bigint.add method

File: presto-main/src/test/java/com/facebook/presto/sql/gen/TestExpressionCompiler.java (hunk 1 above)
Comment (maciejgrzybek) [inline presto-main/src/test/java/com/facebook/presto/sql/gen/TestExpressionCompiler.java:294]: Because of the overflow since `extremeLongs` multiplied by big numbers from `longRights` may cause overflow. In some cases even adding `1` to e.g. `Long.MAX_VALUE` may cause an overflow.
I wanted to avoid adding more logic to the test case (to figure out if this is a case of overflow) and I didn't want to change numbers because these are used in other use cases too.

**Comment by maciejgrzybek** [comment]
@cawallin I talked to @raghavsethi and he said this won't be merged before establishing what to do with `MIN_VALUE`.

File: presto-main/src/main/java/com/facebook/presto/operator/aggregation/LongSumAggregation.java (hunk 2 above)
Comment (maciejgrzybek) [inline presto-main/src/main/java/com/facebook/presto/operator/aggregation/LongSumAggregation.java]: Good point. Changed in 70002783ee132e0331f97d22573ba91c9c6b4e0d.

**Comment by maciejgrzybek** [comment]
No more comments given so closing and opening PR against `Facebook/presto` repository (`issue_4523/2` branch). Corresponding PR can be find here: https://github.com/facebook/presto/pull/4577.
